docker/Dockerfile

```diff
@@ -44,27 +44,6 @@ RUN apt-get update \
         /opt/google-fluentd/embedded/share/gtk-doc \
         /opt/google-fluentd/embedded/share/man \
         /opt/google-fluentd/embedded/share/terminfo \
-    # Remove unused gems.
-    && /opt/google-fluentd/embedded/bin/gem uninstall -ax --force \
-        # Gems for ingesting logs to Treasure Data Cloud.
-        td \
-        td-client \
-        td-logger \
-        fluent-plugin-td \
-        fluent-plugin-td-monitoring \
-        hirb \
-        parallel \
-        ohai \
-        mixlib-cli \
-        mixlib-config \
-        mixlib-log \
-        mixlib-shellout \
-        systemu \
```

Comment:
We can still remove these since they're dependencies that'll be installed (see [Gemfile.lock](https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/Gemfile.lock)).

Reply:
Gemfile.lock theoretically contains build-time dependencies (installed in the Ruby virtual environment), not runtime dependencies (installed in the embedded Ruby location), but I'll confirm later today.

Reply:
Confirmed. None of these gems are present in packages released after #179. The last version to contain these gems was 1.6.12.

Reply:
How'd you run it? When I ran `/opt/google-fluentd/embedded/bin/gem list`, they [showed up](https://gist.github.com/jkschulz/4c07558295320452e634a1b68c5db6d7) for me, but maybe I'm doing something wrong?

Reply:
Interesting. I don't see them in either the package:
```
$ rpm -qlp google-fluentd-1.6.30-1.el7.x86_64.rpm | grep specifications/mixlib
$
```
or the list of local gems:
```
$ /opt/google-fluentd/embedded/bin/gem list mixlib

*** LOCAL GEMS ***


$
```
I do see them in the older package:
```
$ rpm -qlp google-fluentd-1.6.12-1.el7.x86_64.rpm | grep specifications/mixlib
/opt/google-fluentd/embedded/lib/ruby/gems/2.4.0/specifications/mixlib-cli-1.7.0.gemspec
/opt/google-fluentd/embedded/lib/ruby/gems/2.4.0/specifications/mixlib-config-3.0.1.gemspec
/opt/google-fluentd/embedded/lib/ruby/gems/2.4.0/specifications/mixlib-log-3.0.1.gemspec
/opt/google-fluentd/embedded/lib/ruby/gems/2.4.0/specifications/mixlib-shellout-2.4.4.gemspec
```
Maybe your machine had the older package installed at some point, and the upgrade didn't clean up properly? Or was that the machine you used to build google-fluentd?

Reply:
I ran it on the same machine I used to build the package -- does that account for it?

Reply:
Yes, it very likely does. I don't know where Omnibus installs the build requirements, but it's possible that the embedded Ruby has access to the system Ruby packages, especially if you are not running under rvm. My `gem list` command was run on a clean machine with a fresh agent install.

Reply:
Interesting, I'll figure out what went wrong with mine later. lgtm!